compiler/modules/nback/structure.bal

```diff
@@ -324,19 +324,54 @@ function mappingOrderFields(t:MappingAtomicType mat, string[] fieldNames, bir:Op
 function buildMappingGet(llvm:Builder builder, Scaffold scaffold, bir:MappingGetInsn insn) returns BuildError? {
     bir:Register mappingReg = insn.operands[0];
     bir:StringOperand keyOperand = insn.operands[1];
+    llvm:BasicBlock bbExact = scaffold.addBasicBlock();
+    llvm:BasicBlock bbInexact = scaffold.addBasicBlock();
+    llvm:BasicBlock bbJoin = scaffold.addBasicBlock();
+    llvm:Value mapping = builder.load(scaffold.address(mappingReg));
+    llvm:Value isExact = buildIsExact(builder, scaffold, mapping);
+    builder.condBr(isExact, bbExact, bbInexact);
     int? fieldIndex = mappingFieldIndex(scaffold.typeContext(), mappingReg.semType, keyOperand);
-    RuntimeFunction rf;
-    llvm:Value k;
-    if fieldIndex == () {
-        rf = mappingGetFunction;
-        k = check buildString(builder, scaffold, keyOperand);
+    builder.positionAtEnd(bbExact);
+    buildMappingMemberExactGet(builder, scaffold, insn, mapping, fieldIndex);
+    builder.br(bbJoin);
+    builder.positionAtEnd(bbInexact);
+    check buildMappingMemberNonExactGet(builder, scaffold, insn, mapping, fieldIndex);
+    builder.br(bbJoin);
+    builder.positionAtEnd(bbJoin);
+}
+
+function buildMappingMemberExactGet(llvm:Builder builder, Scaffold scaffold, bir:MappingGetInsn insn, llvm:Value mapping, int? index) {
+    bir:Register mappingReg = insn.operands[0];
+    bir:StringOperand keyOperand = insn.operands[1];
+    llvm:Value member;
+    if index == () {
+        member = llvm:constNull(LLVM_TAGGED_PTR);
```

Comment:
@jclark, I am still checking the correctness of this code block. I will check and update this part.